### PR TITLE
Able to show outiline in readonly mode

### PR DESCRIPTION
**Pull request by vinicius73**
* Resolves: #2904 
* Target version: master 

### Readonly

https://user-images.githubusercontent.com/1561347/190261995-20c63bc2-5c7b-49b3-a6e6-5cc34a480f42.mp4

### Public

https://user-images.githubusercontent.com/1561347/190262053-cd9f71de-af29-4735-98a9-4db1ded0d385.mp4

### CODE REVIEW

**Review by max-nextcloud — COMMENTED** [review]
Code looks good.

However if i click an entry in the outline it does not move to that heading. That's not what i'd expect because the entry is highlighted and looks like a link to me.

I also managed to trigger some funny behavior when clicking at the end of headings. I was not able to produce the same behavior on the editable view of the owner of the file.
[Bildschirmaufzeichnung vom 15.09.2022, 08:19:51.webm](https://user-images.githubusercontent.com/97337118/190329639-3aeb07cc-7ab9-45c8-9bf1-4359edc32544.webm)

File: src/components/Menu/ReadonlyBar.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	components: { ActionEntry },
+	setup() {
+		return {
+			visibleEntries: entries,
+		}
+	},
+})
+</script>
+
+<style scoped>
```
Comment (susnux) [inline src/components/Menu/ReadonlyBar.vue:30]: To be consistent with the edit mode the menu bar should be `sticky`.

**Review by jancborchardt — COMMENTED** [review]
Nice! Another point in addition to @susnux:
The "x" close button is a bit floating in the air. Would be better if it’s directly after the "Outline" text instead of floating on the right.

**Comment by mejo-** [comment]
Thanks a lot for looking into this @vinicius73 :heart:

Unfortunately using the menubar space in read-only mode conflicts with Collectives, where we already use this space for Collectives-specific content (last changed user and timestamp, in future also things like count of backlinks and attachments). Sorry, I don't have a quick solution for this. Maybe we should discuss it in our next team call?

**Comment by vinicius73** [comment]
> Thanks a lot for looking into this @vinicius73 :heart:
> 
> Unfortunately using the menubar space in read-only mode conflicts with Collectives, where we already use this space for Collectives-specific content (last changed user and timestamp, in future also things like count of backlinks and attachments). Sorry, I don't have a quick solution for this. Maybe we should discuss it in our next team call?

What do you think about make able to inject a slot that space and expose some editor methods?

So that way you will be able to not only inject more options also access internal editor methods in a safe way

**Comment by vinicius73** [comment]
@jancborchardt what do you think about this?

image

**Comment by mejo-** [comment]
> What do you think about make able to inject a slot that space and expose some editor methods?
> 
> So that way you will be able to not only inject more options also access internal editor methods in a safe way

That would be the way to go indeed :blush:

**Comment by jancborchardt** [comment]
@vinicius73 it's better if the x is on the right if the text, so that the text is left-aligned with the headings below. :) Just not all the way over away from the text.

**Comment by blizzz** [comment]
master is 26, please trigger backport if necessary.

**Comment by vinicius73** [comment]
@mejo- I've rewritten the all PR, check if you can please

**Comment by mejo-** [comment]
Nice, thanks a lot @vinicius73. Works like expected. I removed the `top: 104px` as discussed [here](https://github.com/nextcloud/text/issues/2904#issuecomment-1247244100). It doesn't solve the problem for Collectives yet, but at least it doesn't break Collectives any longer - and it adds outline support for read-only Text shares, which is great :heart_eyes:

**Review by mejo- — APPROVED** [review]

**Comment by mejo-** [comment]
/backport 103d8ef26df9ec6c86098eecbf16304f08248f7d,7d9a18b15c56608fa38492ec9d7f877130922488 to stable25

**Pull request merged**